integration-tests/smallrye-context-propagation/src/test/java/io/quarkus/context/test/ContextEndpoint.java

```diff
@@ -83,8 +76,11 @@ public CompletionStage<String> servletTest(@Context UriInfo uriInfo) {
     public CompletionStage<String> servletThreadContextTest(@Context UriInfo uriInfo) {
         ExecutorService executor = Executors.newSingleThreadExecutor();
         CompletableFuture<String> ret = allTc.withContextCapture(CompletableFuture.completedFuture("OK"));
+        RequestBean instance = Arc.container().instance(RequestBean.class).get();
```

Comment:
Why don't you inject the `RequestBean` in the resource class? It's a `
@RequestScoped` bean so you should get a client proxy anyway...

Reply:
I know but in the end I did it this way because there were several tests already doing it, so it was partly a copy paste.
Explicit resolution might also make it more clear to anyone reading this test who is not aware of internals of Arc and proxies. After all, it is a just a test.
But I can rewrite it if you insist :)